app/uk/gov/hmrc/exports/controllers/DeclarationController.scala

```diff
@@ -48,7 +51,16 @@ class DeclarationController @Inject()(
   def getByID(id: String): Action[AnyContent] = authenticator.authorisedAction(parse.default) { implicit request =>
     declarationService.findOne(id, request.eori.value).map {
       case Some(declaration) => Ok(Json.toJson(declaration))
-      case None => NotFound
+      case None              => NotFound
+    }
+  }
+
+  def deleteByID(id: String): Action[AnyContent] = authenticator.authorisedAction(parse.default) { implicit request =>
+    declarationService.findOne(id, request.eori.value).flatMap {
+      case Some(declaration) if declaration.status != DeclarationStatus.COMPLETE =>
```

Comment:
I would move some of this logic into Service layer. It is not something that Controller needs to care about.

Reply:
It should be even on repository level